runtime/src/bank_forks.rs

```diff
@@ -463,45 +462,40 @@ impl BankForks {
         // part of the same set of `banks` in a single `set_root()` invocation.
         // While (very) unlikely for a validator with default snapshot intervals,
         // it *is* possible, and there are tests to exercise this possibility.
-        let abs_request_interval = self.abs_request_interval();
-        if let Some(bank) = banks.iter().find(|bank| {
-            bank.slot() > self.latest_abs_request_slot
-                && bank.block_height() % abs_request_interval == 0
-        }) {
-            let bank_slot = bank.slot();
-            self.latest_abs_request_slot = bank_slot;
-            squash_timing += bank.squash();
-
-            is_root_bank_squashed = bank_slot == root;
-
-            let mut snapshot_time = Measure::start("squash::snapshot_time");
-            if self.snapshot_config.is_some()
-                && accounts_background_request_sender.is_snapshot_creation_enabled()
-            {
+        if let Some(abs_request_interval) = self.abs_request_interval() {
+            assert!(accounts_background_request_sender.is_snapshot_creation_enabled());
```

Comment:
This seems more strictly correct but it feels less risky to keep this as a conditional. I would prefer to skip a snapshot request over crashing a validator.